snapcraft/plugins/go.py

```diff
@@ -101,7 +102,13 @@ def get_pull_properties(cls):
 
     def __init__(self, name, options, project):
         super().__init__(name, options, project)
-        self.build_packages.append('golang-go')
+
+        # Do not try to install golang if it is already available in our PATH.
+        try:
+            subprocess.check_output(['go', 'version'])
+        except OSError:
+            self.build_packages.append('golang-go')
```

Comment:
@elopio "a little slower" is a bit of an understatement :) Rebuilding my toolchain from source every time I want to package my golang application seems a little bit extreme, I think I'll just maintain a local fork of snapcraft with this patch until a more mature solution emerges.

Thanks for your time!